### DISCUSSION TITLE
Is it possible to add a label for major version upgrades?

### DISCUSSION THREAD

**Discussion by yadavan88** [scala-steward-org/scala-steward#3144] [Q&A] [answered]
Hi Scala-Steward team,
I tried searching for this, but didn't find it. 
I want to check if there is any possibility of marking a PR with a particular label, if it contains major version upgrade. 
The reason is that I want to apply mergify auto-merge rules to all upgrades except major version. If not the label way, is there any specific way to identify if the PR is for major version update or not?

Thanks in advance
Yadu

**Comment by yadavan88** [comment]
Ok, so i went through the code and then realized that there are labels adding to the PR description as early-semver-major and so on. However, these are not added as label in github PR. Is there any config available to add the label to the PR ?

I checked the file `GitHubApiAlg.scala` and the method `createPullRequest` already seems to add the label if present.
```
_ <- F.whenA(data.labels.nonEmpty)(labelPullRequest(repo, pullRequestOut.number, data.labels))
```
I can see the label info in the PR description:
```
 labels: library-update, early-semver-major, semver-spec-major, old-version-remains, commit-count:1 
```
But there are no labels in the PR. I think I am missing out something obvious, could someone help me?

**Reply by seveneves** [reply]
I think labels attached directly to a PR are defined in this configuration section 

```
# pullRequests.customLabels allows to add custom labels to PRs.
# This is useful if you want to use the labels for automation (project board for example).
# Defaults to no labels (no labels are added).
pullRequests.customLabels = [ "dependencies", "scala" ]
```

These are different from labels added to the description of a PR that indicate type of an update 

> labels: library-update, early-semver-major, semver-spec-major, old-version-remains, commit-count:1

**Reply by yadavan88** [reply]
Thanks for the response. I saw these, but these are kind of fixed labels. 

What I was trying to do is to have conditional labels based on the version update. I mean, I want to label only if there is major version update. Scala-Steward is already adding this label to the description (for e.g: early-semver-major), which I want to add to the PR label.

**Reply by seveneves** [reply]
I don't think this is possible. Try renovate bot instead

**Reply by yadavan88** [reply]
Oh.. is it? 
The case class `NewPullRequestData` has the labels and the same labels is used within the description. Also in the `createPullRequest` these labels are also passed by invoking the method `labelPullRequest` as mentioned [here](https://docs.github.com/en/free-pro-team@latest/rest/issues/labels?apiVersion=2022-11-28#add-labels-to-an-issue). 
I am confused what is the reason it is not adding the label still! :(

Al these from the file `GitHubApiAlg.scala`

**Reply by alejandrohdezma** [reply, marked as answer]
Hey @yadavan88, Scala Steward only adds labels to the PR (as GitHub labels, apart from description) if flag [`--add-labels`](https://github.com/scala-steward-org/scala-steward/blob/22261658573e0e68bdeca8520f5da78bd61adbdc/docs/help.md?plain=1#L43) is added when running the CLI. How are you running Scala Steward? If you are using the [GitHub Action](https://github.com/scala-steward-org/scala-steward-action) you can just use the [`other-args`](https://github.com/scala-steward-org/scala-steward-action/blob/82f04a97ecba97f2f930e006bd6ff5f26eadd1ba/action.yml#L116) input (or contribute the `-add-labels` one 😉).

**Reply by yadavan88** [reply]
Thanks @alejandrohdezma 
I am using the github actions. I will check out the links provided. Thanks for the suggestion.

**Reply by yadavan88** [reply]
I went through the links. So, if I am using the public instance of scala-steward, then this config won't be applicable, right? 
If then, I will try using the actions instead and try to configure it. Thanks again.

**Reply by alejandrohdezma** [reply]
Correct, the public instance of Scala Steward won't be able to label your repository, as it lacks the necessary permissions

**Reply by yadavan88** [reply]
Thank you @alejandrohdezma 
Sorry to bother again. Do you know any sample repos that use this github action? I am trying , but unable to make it work fully.

**Reply by yadavan88** [reply]
never mind, it was some silly mistake, works now and label are getting added :) Thank you so much for your help, really appreciate it..
